unordered_associative_task_2/student/src/impl.cpp

```diff
@@ -30,6 +30,31 @@ auto xx_hash = [] (const std::string &file)
 template<class FilterCriteria>
 filtering_map filter(const std::vector<std::string>& files, FilterCriteria filter)
 {
+    filtering_map unique;
+
+    for(auto path : files){
+        std::size_t key;
+
+        if(filter){
+
+            std::ifstream stream {path};
+
+            if (!stream)
+            {
+                std::cerr << "File " << path << " can't be opened" << std::endl;
+            }
+
+            key = (!stream)? 0 : fs::file_size(path);
+        } else {
+
+            key = xx_hash(path);
+        }
+
+        std::vector<std::string> none = {};
+        unique.try_emplace(key, none);
+        unique[key].push_back(path);
```

Comment:
yep, but please 
`for (const auto& file : files)
    {
        filtered[filter(file)].push_back(file);
    }`